src/utils/common/nixl_log.h

```diff
@@ -0,0 +1,84 @@
+/*
+ * SPDX-FileCopyrightText: Copyright (c) 2025 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#ifndef __NIXL_LOG_H
+#define __NIXL_LOG_H
+
+#include "absl/log/log.h"
+#include "absl/log/check.h"
+#include "absl/log/initialize.h"
+
+/*-----------------------------------------------------------------------------*
+ * Logging Macros (Abseil Stream-style)
+ *-----------------------------------------------------------------------------*
+ * Ordered by severity (highest to lowest)
+ * Usage: NIXL_INFO << "Message part 1 " << variable << " message part 2";
+ */
+
+/*
+ * Logs a message and terminates the program unconditionally.
+ * Maps to Abseil LOG(FATAL). Use for unrecoverable errors.
+ */
+#define NIXL_FATAL LOG(FATAL)
+
+/* Logs messages unconditionally (maps to Abseil ERROR level) */
+#define NIXL_ERROR LOG(ERROR)
+
+/*
+ * Logs messages unconditionally (maps to Abseil WARNING level)
+ * Stripped from release buids.
+ */
+#define NIXL_WARN LOG(WARNING)
+
+/*
+ * Logs messages unconditionally (maps to Abseil INFO level)
+ * Stripped from release buids.
```

Comment:
pls make it available in release builds. But the default log level should be warn anyway